### ISSUE TITLE
Mypy incorrectly warns when passing literals as kwarg keywords

### ISSUE THREAD

**Issue by nipunn1313** [python/mypy#10023]
**Bug Report**

```
from typing import Any, Dict, Literal
kw: Dict[Literal["a", "b"], Any]
def func(a, b):
    pass
func(**kw)
```
Leads to
```
main.py:8: error: Keywords must be strings
```

https://mypy-play.net/?mypy=latest&python=3.9&gist=582cf5e8c0a6e6a5881d391da64e1455

**Expected Behavior**
Expect this to be allowable - as these literals count as strings

**Comment by JukkaL** [comment]
@JelleZijlstra Yes. 

This should be fine, I think:
```py
from typing import Any, Dict, Literal
kw: Dict[Literal["a", "b"], Any]
def func(a=None, b=None):
    pass
func(**kw)
```

**Comment by vickyhuo** [comment]
Hey @JukkaL I want to work on this. Can you please assign it to me?

**Comment by JukkaL** [comment]
@vickyhuo Done!

**Comment by Uttam-Singhh** [comment]
Can I work on it in the future?

**Comment by Soumokanti123** [comment]
> @JelleZijlstra Yes.
> 
> This should be fine, I think:
> 
> ```python
> from typing import Any, Dict, Literal
> kw: Dict[Literal["a", "b"], Any]
> def func(a=None, b=None):
>     pass
> func(**kw)
> ```

still it having a problem

**Comment by vickyhuo** [comment]
Hi @kamilturek, I believe that is the problem as well. I'm currently still working on a fix

**Comment by hauntsaninja** [comment]
Sounds like this is just another symptom of Mapping not being covariant in its key type

**Comment by kamilturek-sc** [comment]
@hauntsaninja What's the reason of this?

**Comment by kamilturek** [comment]
@hauntsaninja What's the reason of this?

**Comment by hauntsaninja** [comment]
It's unsafe, see https://github.com/python/typing/issues/445

**Comment by jp-larose** [comment]
Reading through this issue, I was trying to figure out if the issue was that `kw` wasn't assigned any values in the sample code.  However, I can validate that even with a `dict` assigned to `kw`, the issue persists, i.e. the same error message is output by mypy.  Additionally, PyCharm gives a hint when assigning the value as follows.

```python
from typing import Any, Dict, Literal
kw: Dict[Literal["a", "b"], Any]

kw = {"a": 1, "b": 2}  # Expected type 'dict[Literal["a", "b"], Any]', got 'dict[str, int]' instead 

def func(a, b):
    pass

func(**kw)
```

Seems to echo @kamilturek's discovery in the debugger.

**Comment by pulkit-30** [comment]
Hi,
Is anyone working on this issue?
If not, please assign me this issue, and please help me out with some references from where I can start digging into it.

**Comment by nipunn1313** [comment]
Looks like #10237 is a PR trying to solve this 

Perhaps one way to help is to help review 10237 - by reading the code - and checking it out and running it to double check that it fixes the issue. I know the  maintainers appreciate when folks do code review.

In case you're not familiar - github renders this in the header with "may be fixed by" -
![image](https://user-images.githubusercontent.com/1300387/136252707-3bb98fdb-1b6b-45df-9e6e-91a59297a9f9.png)

**Comment by AaronDMarasco** [comment]
#13674 may be a dupe as well
